include/oneapi/mkl/dft/detail/mklcpu/onemkl_dft_mklcpu.hpp

```diff
@@ -28,116 +28,16 @@
-    /*USM version*/                                                                                     \
-                                                                                                        \
-    /*In-place transform*/                                                                              \
-    sycl::event compute_backward_usm_inplace_##EXT(                                                     \
-        descriptor<PRECISION, DOMAIN> &desc, T_BACKWARD *inout,                                         \
-        const std::vector<sycl::event> &dependencies = {});                                             \
-                                                                                                        \
-    /*In-place transform, using config_param::COMPLEX_STORAGE=config_value::REAL_REAL data format*/     \
-    sycl::event compute_backward_usm_inplace_split_##EXT(                                               \
-        descriptor<PRECISION, DOMAIN> &desc, T_REAL *inout_re, T_REAL *inout_im,                        \
-        const std::vector<sycl::event> &dependencies = {});                                             \
-                                                                                                        \
-    /*Out-of-place transform*/                                                                          \
-    sycl::event compute_backward_usm_outofplace_##EXT(                                                  \
-        descriptor<PRECISION, DOMAIN> &desc, T_BACKWARD *in, T_FORWARD *out,                            \
-        const std::vector<sycl::event> &dependencies = {});                                             \
-                                                                                                        \
-    /*Out-of-place transform, using config_param::COMPLEX_STORAGE=config_value::REAL_REAL data format*/ \
-    sycl::event compute_backward_usm_outofplace_split_##EXT(                                            \
-        descriptor<PRECISION, DOMAIN> &desc, T_REAL *in_re, T_REAL *in_im, T_REAL *out_re,              \
-        T_REAL *out_im, const std::vector<sycl::event> &dependencies = {});
-
-ONEAPI_MKL_DFT_BACKEND_SIGNATURES(f, precision::SINGLE, domain::REAL, float, float,
-                                  std::complex<float>)
-ONEAPI_MKL_DFT_BACKEND_SIGNATURES(c, precision::SINGLE, domain::COMPLEX, float, std::complex<float>,
-                                  std::complex<float>)
-ONEAPI_MKL_DFT_BACKEND_SIGNATURES(d, precision::DOUBLE, domain::REAL, double, double,
-                                  std::complex<double>)
-ONEAPI_MKL_DFT_BACKEND_SIGNATURES(z, precision::DOUBLE, domain::COMPLEX, double,
-                                  std::complex<double>, std::complex<double>)
-
-#undef ONEAPI_MKL_DFT_BACKEND_SIGNATURES
+ONEMKL_EXPORT oneapi::mkl::dft::detail::commit_impl* create_commit(sycl::queue queue, dft_values values);
```

Comment:
This was done to avoid circular dependency. (Tadej suggests forward declaring them)
Anant says the UML diagram will prevent this. 
Anant to try out the way Tadej mentioned and get back if that causes issue.

Reply:
initial setup : 
descriptor.hpp was included in onemkl_dft_mklgpu.hpp
following the UML I need onemkl_dft_mklgpu.hpp in descriptor (latter are backend specific) which are in itself forward declarations to the implimentation is src. (conforms to many implimentation in other domains, e.g. rng)

reverting this to the original stubs seems counter intutive because from descriptor.hpp how can we have a call to methods in onemkl_dft_mklxpu.hpp without including the forward declaration thereby inducing a circular dependency.
maybe i am being short sighted here but correct me if you think there is merit to the original stubs.

secondly, 
on you point about having a templatized create_commit(queue, values)
1. I need the set_value information again in the instance of commit_derived_impl to setup device specific set_value, for which I need to pas the value_struct.
2. given the above point, would it be straight-forward and slightly advantageous to have the domain, precision be passed in the struct, rather that it be templated.

Reply:
The easiest way to resolve circular dependencies here is to create `descriptor.cpp` file and move implementations of various `commit` overloads there, leaving only declarations in the `descriptor.hpp` file. Then you won't need to include `onemkl_dft_mklgpu.hpp` in the `descriptor.hpp`.

1. commit_derived_impl can be templated for the compile time resolution. For runtime resolution you can create 4 functions for all possible combinations of precision and domain (the way I proposed in the original design with `commit_##EXT`). There is no need to put template parameters into values struct.

2. It might be simpler now, but once you need to construct the descriptor for the backend you will need those as templates again. Then you will have to use something like switch statements to determine which combination of templates you want to use. This will just add unnecessary overhead for something that could be dealt with at compile time. Leaving them as templates (or different functions for the parts that need to be written in C) is better design.

Reply:
these forward declarations were added in commit 6cf88ba4eaeba1d2be10fb89c47777059955dd2f, and thi implimentation moved to descriptor.cpp